plugins/destination/firehose/client/spec.go

```diff
@@ -7,16 +7,35 @@ import (
 )
 
 type Spec struct {
-	StreamARN  string `json:"stream_arn,omitempty"`
+	StreamARN  string `json:"stream_arn"`
 	NoRotate   bool   `json:"no_rotate,omitempty"`
 	MaxRetries *int   `json:"max_retries,omitempty"`
+
+	MaxRecordSizeBytes int `json:"max_record_size_bytes,omitempty"`
+	MaxBatchRecords    int `json:"max_batch_records,omitempty"`
+	MaxBatchSizeBytes  int `json:"max_batch_size_bytes,omitempty"`
 }
 
+const (
+	defaultMaxRecordSizeBytes = 1024000
+	defaultMaxBatchRecords    = 500
+	defaultMaxBatchSizeBytes  = 4194000
+)
+
 func (s *Spec) SetDefaults() {
 	if s.MaxRetries == nil {
 		s.MaxRetries = new(int)
 		*s.MaxRetries = 5
 	}
+	if s.MaxRecordSizeBytes == 0 {
```

Comment:
This cheeky `== 0` allows to use -1 max record size for no limit.

Reply:
Why would you want no limit? I don't think it will work if you send anything larger than 1MB to Kinesis https://aws.amazon.com/kinesis/data-firehose/faqs/

> The maximum size of a record (before Base64-encoding) is 1024 KB if your data source is Direct PUT or Kinesis Data Streams.

Reply:
Ok, agreed. Btw it says before the base64 encoding, so I don't think the limiting code is counting the bytes correctly.